Packet++/src/TcpReassembly.cpp

```diff
@@ -473,34 +473,40 @@ void TcpReassembly::handleFinOrRst(TcpReassemblyData* tcpReassemblyData, int8_t
 
 void TcpReassembly::checkOutOfOrderFragments(TcpReassemblyData* tcpReassemblyData, int8_t sideIndex, bool cleanWholeFragList)
 {
+	if (m_ProcessingOutOfOrder)
+	{
+		return;
+	}
+
+	m_ProcessingOutOfOrder = true;
 	bool foundSomething = false;
 
 	do
 	{
 		PCPP_LOG_DEBUG("Starting first iteration of checkOutOfOrderFragments - looking for fragments that match the current sequence or have smaller sequence");
 
-		int index = 0;
+		PointerVector<TcpFragment>::VectorIterator it;
 		foundSomething = false;
 
 		do
 		{
-			index = 0;
+			it = tcpReassemblyData->twoSides[sideIndex].tcpFragmentList.begin();
```

Comment:
We could probably cache the `tcpReassemblyData->twoSizes[sideIndex]` result to a reference variable? It has been accessed 18 times in the function.

Reply:
Update in this commit: https://github.com/seladb/PcapPlusPlus/pull/1346/commits/d1b610d27ec78704539fb6f7f7119b404c8b3e89